mgr/src/certs/lightning.rs

```diff
@@ -103,19 +106,46 @@ fn create_or_update_ca_cert(
 }
 
 fn create_or_update_cert(
-    key_path: &Path,
-    cert_path: &Path,
+    cert_dir: &Path,
     ca_key_path: &Path,
     ca_cert_path: &Path,
     policy: &CertRenewPolicy,
+    host: &Host,
 ) -> Result<()> {
-    if cert_path.exists() && cert_is_atleast_valid_for(cert_path, policy.cert_renew_seconds) {
+    let key_path = cert_dir.join(format!("{}.key", host.name));
+    let cert_path = cert_dir.join(format!("{}.pem", host.name));
+    let ip_hash_path = cert_dir.join(format!("{}-ip.hash", host.name));
+
+    let ip_hash = calculate_hash_for_address_in_san(
+        host.api_ip_access_list.is_empty(),
+        vec![&host.ipv4_address, &host.ipv6_address],
+    );
```

Comment:
Rather than hashing you can join strings here i.e. with a ","
The hash does not really add any value and makes debugging harder.

Reply:
In theory you can also extract certificate names like this:

```
$ openssl x509 -noout -text -certopt no_subject,no_header,no_version,no_serial,no_signame,no_validity,no_issuer,no_pubkey,no_sigdump,no_aux -ext subjectAltName -in db-01.pem
        X509v3 extensions:
            X509v3 Subject Alternative Name:
                DNS:localhost, IP Address:127.0.0.1, IP Address:0:0:0:0:0:0:0:1
X509v3 Subject Alternative Name:
    DNS:localhost, IP Address:127.0.0.1, IP Address:0:0:0:0:0:0:0:1
```

Would require some simple rust parsing though. I guess just having a separate files is a bit less code?

Reply:
I agree with you.  Directly checking on the certificate will be straightforward to know if we need to update it or not.